src/main/java/org/datadog/jenkins/plugins/datadog/configuration/api/key/DatadogCredentialsApiKey.java

```diff
@@ -0,0 +1,193 @@
+package org.datadog.jenkins.plugins.datadog.configuration.api.key;
+
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import com.cloudbees.plugins.credentials.domains.URIRequirementBuilder;
+import hudson.Extension;
+import hudson.model.Descriptor;
+import hudson.model.Item;
+import hudson.security.ACL;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import hudson.util.Secret;
+import java.util.Collections;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.plaincredentials.StringCredentials;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+@Symbol("datadogCredentialsApiKey")
+public class DatadogCredentialsApiKey extends DatadogApiKey {
+
+    private final String credentialsId;
+
+    @DataBoundConstructor
+    public DatadogCredentialsApiKey(String credentialsId) {
+        this.credentialsId = credentialsId;
+    }
+
+    public String getCredentialsId() {
+        return credentialsId;
+    }
+
+    @Override
+    public Secret getKey() {
+        StringCredentials credential = getCredentialFromId(credentialsId);
+        return credential != null ? credential.getSecret() : null;
+    }
+
+    /**
+     * Gets the StringCredentials object for the given credential ID
+     *
+     * @param credentialId - The ID of the credential to get
+     * @return a StringCredentials object
+     */
+    static StringCredentials getCredentialFromId(String credentialId) {
+        if (StringUtils.isBlank(credentialId)) {
+            return null;
+        }
+        return CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentials(
+                        StringCredentials.class,
+                        Jenkins.get(),
+                        ACL.SYSTEM,
+                        URIRequirementBuilder.fromUri(null).build()),
+                CredentialsMatchers.allOf(CredentialsMatchers.withId(credentialId))
+        );
```

Comment:
I couldn't find any. Anyway, I don't think that suppressing the error is better than propagating it in this case. If credentials cannot be retrieved, the error will eventually be propagated to `DatadogGlobalConfiguration#configure` where it will be logged and updating the config will be aborted.